pkg/test/provisioner.go

```diff
@@ -73,6 +74,9 @@ func Provisioner(overrides ...ProvisionerOptions) *v1alpha5.Provisioner {
 		})
 	}
 
+	raw := &runtime.RawExtension{}
+	ExpectWithOffset(1, raw.UnmarshalJSON(lo.Must(json.Marshal(options.Provider)))).To(Succeed())
```

Comment:
Why is this necessary?

Reply:
Provider is an interface coming from the options and isn't typed as `runtime.RawExtension` so it needs to be converted to that type before assignment